### PR TITLE
Update Typescript to 3.1

### PR DESCRIPTION
**Pull request by sirDonovan**

### CODE REVIEW

File: chat-plugins/info.js [hunk 1]
```diff
@@ -12,10 +12,7 @@
 
 'use strict';
 
-/** @typedef {(this: CommandContext, target: string, room: BasicChatRoom, user: User, connection: Connection, cmd: string, message: string) => (void)} ChatHandler */
-/** @typedef {{[k: string]: ChatHandler | string | true | string[]}} ChatCommands */
-
-/** @type {ChatCommands} */
+/** @type {import('./../chat').ChatCommands} */
```
Comment (Zarel) [inline chat-plugins/info.js]: !!!!!!!!!!!!!!!

THIS WORKS NOW??????

Comment (sirDonovan) [reply chat-plugins/info.js]: Yeah :D

Comment (Zarel) [reply chat-plugins/info.js]: I wonder if we can globalize this, then? Can `import('./../chat').ChatCommands` be stuck in `globals.ts` or something?

Comment (sirDonovan) [reply chat-plugins/info.js]: That's a good idea, I'll try it.

Comment (sirDonovan) [reply chat-plugins/info.js]: Updated with that.

**Comment by Zarel** [comment]
Can you do the same for PageTable?

**Comment by sirDonovan** [comment]
Did the same for that.

**Comment by Zarel** [comment]
Okay, I'll approve once I can review on my other computer; apparently 600ish changes is too laggy for this computer.

**Comment by Zarel** [comment]
Looks like a lot of objects that didn't need to be typed before need to be typed now? Sounds like possibly a TypeScript bug.

**Comment by sirDonovan** [comment]
@Zarel yeah it's because in strict mode, objects without index signatures are treated as if they had <code>{[k: string]: any}</code>, which triggers <code>noImplicitAny</code> errors whenever you access them.

**Pull request merged**